doc/api-design.yml

```diff
@@ -0,0 +1,903 @@
+openapi: 3.0.3
+info:
+  title: Quizzard API
+  description: API documentation for the Quizzard application.
+  version: 1.0.0
+servers:
+  - url: http://localhost:8080/api
+    description: Development server
+components:
+  securitySchemes:
+    bearerAuth:
+      type: http
+      scheme: bearer
+      bearerFormat: JWT
+  schemas:
+    User:
+      type: object
+      properties:
+        id:
+          type: integer
+        username:
+          type: string
+        name:
+          type: string
+        email:
+          type: string
+          format: email
+        profile_picture:
+          type: string
+          format: uri
+        english_proficiency:
+          type: string
+          enum: [a1,a2,b1,b2,c1,c2]
+        points:
+          type: integer
+        created_quizzes:
```

Comment:
Is this the total number of quizzes created by the user or ids for all of the quizzes created by the user? If latter, doesn't it make sense to make this field an array?

Reply:
it is total number of quizzes created.